service/controller/app/v1/resource/configmap/current.go

```diff
@@ -46,43 +45,34 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding configmap %#q in namespace %#q", name, r.chartNamespace))
 
-	ch := make(chan response)
+	ch := make(chan struct{})
 
-	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
-	defer cancel()
+	var configmap *corev1.ConfigMap
 
 	go func() {
-		configmap, err := cc.K8sClient.CoreV1().ConfigMaps(r.chartNamespace).Get(name, metav1.GetOptions{})
-
-		ch <- response{
-			Configmap: configmap,
-			Error:     err,
-		}
+		configmap, err = cc.K8sClient.CoreV1().ConfigMaps(r.chartNamespace).Get(name, metav1.GetOptions{})
+		close(ch)
 	}()
 
-	var res = response{}
-
 	select {
-	case res = <-ch:
+	case <-ch:
 		// Fall through.
-	case <-ctx.Done():
-		if errors.Is(ctx.Err(), context.DeadlineExceeded) {
-			// Set status so we don't try to connect to the tenant cluster
-			// again in this reconciliation loop.
-			cc.Status.TenantCluster.IsUnavailable = true
-
-			r.logger.LogCtx(ctx, "level", "debug", "message", "timeout getting configmap")
-			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-			resourcecanceledcontext.SetCanceled(ctx)
-			return nil, nil
-		}
+	case <-time.After(3 * time.Second):
```

Comment:
Seems a bit random to wait for 3 seconds - any reason for this specifically?

Reply:
To get TenantClusterUnavailable error in helmclient, it usually takes more than 30 seconds. By this change, we would only need 3 seconds to cancel it.

Reply:
Yes 30 seconds is too long because the operator needs to process other CRs. We can iterate on this if we need to reduce it further.

Reply:
My question would be why we wait at all instead of doing it in the next reconciliation period - not certain if that is better though.

I'll think about this.